config/config.go

```diff
@@ -1,25 +1,110 @@
 package config
 
 import (
+	"log"
 	"net/http"
 	"time"
+
+	"github.com/spf13/viper"
 )
 
 // Config holds configuration settings for establishing a connection and handling
 // request details in the application.
 type Config struct {
-	Addr      string            // The base address of the SPV Wallet API.
-	Timeout   time.Duration     // The HTTP requests timeout duration.
+	Addr      string            `mapstructure:"addr"`    // The base address of the SPV Wallet API.
+	Timeout   time.Duration     `mapstructure:"timeout"` // The HTTP requests timeout duration.
 	Transport http.RoundTripper // Custom HTTP transport, allowing optional customization of the HTTP client behavior.
 }
 
-// NewDefaultConfig returns a default configuration for connecting to the SPV Wallet API,
-// setting a one-minute timeout, using the default HTTP transport, and applying the
-// base API address as the addr value.
-func NewDefaultConfig(addr string) Config {
-	return Config{
-		Addr:      addr,
-		Timeout:   1 * time.Minute,
-		Transport: http.DefaultTransport,
+// setDefaultValues assigns default values to fields that are not explicitly set.
+func (cfg *Config) setDefaultValues() {
+	if cfg.Addr == "" {
```

Comment:
WDYT about creating file called:
`defaults.go`
and there prepare method:
"GetDefaultConfig" setting necessary fields with default values?
We can specify some reusable consts in this file, especially for 
SPV Wallet default address.